tutorials/migrating-rails-app-from-heroku-to-google-appengine.md

```diff
@@ -0,0 +1,418 @@
+
+1.  Secure your new database instance by setting a password on the default
+    postgres user:
+
+        gcloud sql users set-password postgres no-host \
+          --instance=mydbinstance --prompt-for-password
+
+      When prompted, enter a password for the database.
+
+1.  Create a database:
+
+        gcloud sql databases create mydb --instance=mydbinstance
+
+Currently, your database is owned by the default user, `postgres`. You can
+change the ownership to an unprivileged user later, but for now you should
+keep the default user to ensure that the data migration goes smoothly.
+
+### Connect to your Cloud SQL instance
+
+By default, Cloud SQL instances are secured: to connect using standard
+Postgres protocols, you are required to whitelist your your IP address.
+This security measure, while necessary, makes it challenging to establish
+_ad hoc_ database connections.
+
+Cloud SQL provides the [Cloud SQL Proxy](https://cloud.google.com/sql/docs/postgres/sql-proxy)
+command-line tool, which opens a database endpoint that you can connect to locally and which
+proxies back to the database instance via a secure API.
+
+To set up Cloud SQL Proxy, perform the following steps:
+
+1.  [Enable the Google Cloud SQL API in the Cloud Console](https://console.cloud.google.com/apis/library/sqladmin.googleapis.com/).
+
+1.  [Install Cloud SQL Proxy](https://cloud.google.com/sql/docs/postgres/sql-proxy#install).
+    Ensure that `cloud_sql_proxy` is in your environment's `PATH` variable.
+
+1.  The proxy creates local sockets connected to your database instances. In your Rails app
+    directory, create a directory to hold those sockets:
+
+        cd /path/to/my/rails/app
+        mkdir -p tmp/cloudsql
+
+1.  To start the proxy, run the following command:
+
+        cloud_sql_proxy -dir=tmp/cloudsql
+
+    Note: This runs the proxy in the foreground, so subsequent commands
+    need to be run in a separate shell. If you prefer, feel free to
+    background the process instead.
+
+5.  The proxy opens a socket in the directory
+    `/path/to/my/rails/app/tmp/cloudsql/[CONNECTION-NAME]/`. You can then
+    connect to the database simply by pointing at that socket. If you like, you
+    can test connecting with `psql:`
+
+        psql -h /path/to/my/rails/app/tmp/cloudsql/[CONNECTION-NAME] \
+          -U postgres
+
+### Copy data from Heroku to Cloud SQL
+
+Now that you have a way to connect directly to your database instance, you can
+copy the data over from your Heroku database.
+
+1.  Export the contents of the Heroku database:
+
+        heroku pg:backups:capture
+        heroku pg:backups:download
+
+    Those commands create a file `latest.dump` in the current working directory.
+
+1.  Restore the data to your Cloud SQL database. Use the socket provided by
+    the proxy to connect to the database instance:
+
+        pg_restore --verbose --clean --no-acl --no-owner -d mydb -U postgres \
```

Comment:
This command isn't quite working for me. I tried the full path to the proxy and it just stalls without receiving a connection. Full path used:

pg_restore  --verbose --clean --no-acl --no-owner -d mydb -U postgres   -h /Users/franknatividad/ruby-getting-started/tmp/cloudsql/spec-test-ruby-samples:us-central1:mydbinstance/.s.PGSQL.5432

The full command is a bit daunting. What if instead of using the cloudsql proxy. 

0. pg_restore the latest.dump locally.
pg_restore latest.dump > restored_latest.dump
0. Import using the CloudSQL Cloud Console UI